app/models/cms/section.rb

```diff
@@ -201,6 +202,10 @@ def not_own_child
     end
   end
 
+  def system_name_presence
+    errors.add(:system_name, :blank) if system_name.blank? && title.present?
+  end
```

Comment:
What I mean is not to allow submitting a form with an empty `title` instead of later reporting the error. It doesn't make sense to me to design the object around the UI. For the AR object, we want to validate `system_name` is not null. But instead we write the validation around a UI deficiency.

Reply:
OK, I left the system_name validation as it was, and added a html attribute to make the title required in the UI. Unfortunately, the `required` attribute doesn't prevent sending a title filled with only spaces, but it seems that's how it's done in all the project. @josemigallas, @lvillen is that right?

Reply:
TL;DR: Yes, that's right.

`require` prevents the field to be submitted _empty_ of characters, and a blank space is still a character. IMHO this should be up to the controller to validate. If we want some form of client validation, I suggest a hint is added to the field, something like "use a valid name" (Whatever that means).
Optionally, we can specify a `pattern` that the input's value must match. However, this is not an ideal UX without a proper hint / warning, it will only say "The value must match the pattern" or something like that.
It is also possible to set a `min` and `max` length to the input, but I don't think this helps.

The last resort is of course JS. Everything can be done with JS 😄.

Reply:
I think adding the required attribute is enough for this case. It will return a validation error from rails in the case of sending an empty space